roaring/roaring_test.go

```diff
@@ -130,6 +130,46 @@ func TestBitmap_Union(t *testing.T) {
 	}
 }
 
+func TestBitmap_Xor_ArrayArray(t *testing.T) {
+	bm0 := roaring.NewBitmap(0, 1000001, 1000002, 1000003)
+	bm1 := roaring.NewBitmap(0, 50000, 1000001, 1000002)
+	result := bm0.Xor(bm1)
+	if n := result.Count(); n != 2 {
+		t.Fatalf("unexpected n: %d", n)
+	}
+}
+
+func TestBitmap_Xor_ArrayBitmap(t *testing.T) {
+	bm0 := roaring.NewBitmap(1, 70, 200, 4097, 4098)
+	bm1 := roaring.NewBitmap()
+	for i := uint64(0); i < 10000; i += 2 {
+		bm1.Add(i)
+	}
+
+	result := bm0.Xor(bm1)
+	if n := result.Count(); n != 4999 {
+		t.Fatalf("unexpected n: %d", n)
+	}
+}
+
+func TestBitmap_Xor_BitmapBitmap(t *testing.T) {
```

Comment:
I'd like to see the case tested where the two bitmaps have the same bits as well.